### PR TITLE
Login block: link to subscription management page with site selected

### PR DESCRIPTION
**Pull request by simison**
## Proposed changes:

* Link to subscription management with site URL as search param, making it easier to find correct subscription to manage.

Screenshot 2024-03-05 at 19 41 10

Screenshot 2024-03-05 at 19 41 04

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* With member login block on the site
* Login
* See manage -link

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php [hunk 1]
```diff
@@ -126,10 +126,15 @@ function render_block( $attributes ) {
 	}
 
 	if ( Jetpack_Memberships::is_current_user_subscribed() ) {
+		$subscription_id             = Abstract_Token_Subscription_Service::get_token_property( 'sub' );
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php]: Needs more work to be refactored so that this would work.

Comment (simison) [reply projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php]: Having something like `get_subscription_id()` would probably be better?

File: projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php [hunk 2]
```diff
@@ -129,7 +129,7 @@ function render_block( $attributes ) {
 		return sprintf(
 			$block_template,
 			get_block_wrapper_attributes(),
-			'https://wordpress.com/read/subscriptions',
+			'https://wordpress.com/read/subscriptions?s=' . get_site_url(),
```
Comment (pkuliga) [inline projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php:132]: Let's encode the query param.

```suggestion
			'https://wordpress.com/read/subscriptions?s=' . rawurlencode( get_site_url() ),
```

**Review by pkuliga — APPROVED** [review]
Looks good. Added one suggestion.

**Pull request merged**